File: web/deploy.mjs
import { createClient } from "genlayer-js";
import { studioDevnet } from "genlayer-js/chains";
import fs from "fs";

async function main() {
  const keystoreData = JSON.parse(fs.readFileSync("../keystore.json", "utf-8"));
  
  const client = createClient({
    chain: studioDevnet,
    account: {
        keystore: { ...keystoreData, address: '0x' + keystoreData.address },
        password: "12345678"
    }
  });

  const contractCode = fs.readFileSync("../contracts/shiplock.py", "utf-8");

  console.log("Deploying contract...");
  try {
      const tx = await client.deployContract({
        code: contractCode,
        args: [],
        fees: { feeValue: "1000000000000000000" } // 1 GEN
      });
      console.log("Tx hash:", tx.hash);
      
      const receipt = await client.waitForReceipt({ hash: tx.hash });
      console.log("Receipt:", receipt);
      console.log("Contract Address:", receipt.contractAddress);
  } catch(e) {
      console.error(e);
  }
}
main();

File: deploy.py
import json
import os
from eth_account import Account

# Decrypt keystore
with open("keystore.json") as f:
    keystore = json.load(f)

private_key = Account.decrypt(keystore, "12345678").hex()

from genlayer import Genlayer
from genlayer import TransactionFees

client = Genlayer(
    rpc_url="https://studio-dev.genlayer.com/api",
    private_key=private_key
)

with open("contracts/shiplock.py") as f:
    contract_code = f.read()

print("Deploying...")
# Genlayer python SDK
tx_hash = client.deploy_contract(
    contract_code=contract_code,
    args=[],
    fees=TransactionFees(fee_value=1000000000000000000)
)
print(f"Hash: {tx_hash}")

receipt = client.wait_for_receipt(tx_hash)
print(f"Contract Address: {receipt.contract_address}")
